internal/channel/channel.go

```diff
@@ -20,109 +18,140 @@ type Channel struct {
 	Config string `db:"config" json:"-"` // excluded from JSON config dump as this may contain sensitive information
 
 	Logger *zap.SugaredLogger
-	cmd    *exec.Cmd
-	rpc    *rpc.RPC
-	mu     sync.Mutex
-}
 
-func (c *Channel) Start(pluginDir string) error {
-	c.mu.Lock()
-	defer c.mu.Unlock()
+	newConfigCh    chan struct{}
+	stopPluginCh   chan struct{}
+	notificationCh chan Req
+}
 
-	if c.cmd != nil && c.rpc.Err() == nil {
-		return nil
-	}
+type Req struct {
+	req *plugin.NotificationRequest
+	out chan<- error
+}
 
-	if matched, _ := regexp.MatchString("^[a-zA-Z0-9]*$", c.Type); !matched {
-		return fmt.Errorf("channel type must only contain a-zA-Z0-9, %q given", c.Type)
-	}
+func (c *Channel) Start(logger *zap.SugaredLogger) {
+	c.Logger = logger
+	c.newConfigCh = make(chan struct{})
+	c.stopPluginCh = make(chan struct{})
+	c.notificationCh = make(chan Req, 1)
 
-	path := filepath.Join(pluginDir, c.Type)
+	go c.runPlugin()
+}
 
-	cmd := exec.Command(path)
+func (c *Channel) initPlugin() *Plugin {
+	c.Logger.Debug("Initializing channel plugin")
 
-	writer, err := cmd.StdinPipe()
+	p, err := NewPlugin(c.Type, c.Logger)
 	if err != nil {
-		return fmt.Errorf("failed to create stdin pipe: %w", err)
+		c.Logger.Errorw("Failed to initialize channel plugin", zap.Error(err))
+	} else if err := p.SetConfig(c.Config); err != nil {
+		c.Logger.Errorw("Failed to set channel plugin config", err)
+		go terminate(p)
+		p = nil
 	}
```

Comment:
Just `return nil` from the individual error cases instead of reassigning the variable just to return it (or relying on `NewPlugin()` t return nil in this case).